src/viadot/sources/smb.py

```diff
@@ -0,0 +1,330 @@
+"""SMB (file-sharing protocol) connector."""
+
+from pathlib import Path
+
+import pendulum
+from pydantic import BaseModel, root_validator
+import smbclient
+
+from viadot.config import get_source_credentials
+from viadot.exceptions import CredentialError
+from viadot.orchestration.prefect.utils import DynamicDateHandler
+from viadot.sources.base import Source
+
+
+class SMBCredentials(BaseModel):
+    username: str  # username (e.g username@tenant_name.com)
+    password: str
```

Comment:
Use `SecretStr` type, and then you can retrieve value with `get_secret_value()`, see eg. https://github.com/dyvenia/viadot/blob/21ac2c69a88c2582b8b7cd549eb53c5392d57c81/src/viadot/sources/business_core.py#L82

Reply:
[fe128c9](https://github.com/dyvenia/viadot/pull/1127/commits/fe128c91ecaeee5aba9ee40b9b9c35c42c4afc7b)
[4cfbdb6](https://github.com/dyvenia/viadot/pull/1127/commits/4cfbdb6e57a58e187bdfee0190ba0c94b32b3fc3)